### PR TITLE
add Viessmann thermostat

### PR DESCRIPTION
**Pull request by ChrisChoke**
add Viessmann Vicare Radiator Thermostat Valve.

it supports local_temperature, occupied_heating_setpoint, battery, pi_heating_demand, linkquality.

i will analyze it further to look if it supports more data.

its my first pull request. i hope i make it right ;-)

### CODE REVIEW

File: devices.js [hunk 1]
```diff
@@ -9586,6 +9586,28 @@ const devices = [
...
     },
+    //Viessmann
+    {
+        zigbeeModel: ['7637434'],
+        model: 'ZK03840',
+        vendor: 'Viessmann',
+        description: 'Vicare Radiator Thermostat Valve',
+        supports: 'Temperature, Battery',
+        fromZigbee: [fz.thermostat_att_report, fz.battery],
+        toZigbee: [tz.thermostat_occupied_heating_setpoint, tz.thermostat_local_tmperature_calibration],
```
Comment (sunlix) [inline devices.js]: just found a typo in your toZigbee property. <3

```diff
- tz.thermostat_local_tmperature_calibration
+ tz.thermostat_local_temperature_calibration
```

File: devices.js [hunk 2]
```diff
@@ -9586,6 +9586,28 @@ const devices = [
...
+    //Viessmann
+    {
+        zigbeeModel: ['7637434'],
+        model: 'ZK03840',
+        vendor: 'Viessmann',
+        description: 'Vicare Radiator Thermostat Valve',
+        supports: 'Temperature, Battery',
+        fromZigbee: [fz.thermostat_att_report, fz.battery],
+        toZigbee: [tz.thermostat_occupied_heating_setpoint, tz.thermostat_local_temperature_calibration],
+        meta:{configureKey: 1},
```
Comment (sunlix) [inline devices.js:9598]:
```diff
-meta:{configureKey: 1},
+meta: {configureKey: 1},
```

File: devices.js [hunk 3]
```diff
@@ -9586,6 +9586,28 @@ const devices = [
         description: 'RGBW LED bulb with dimmer',
         extend: generic.light_onoff_brightness_colortemp_colorxy,
     },
+    //Viessmann
```
Comment (sunlix) [inline devices.js:9589]:
```diff
-//Viessmann
+// Viessmann
```

File: devices.js [hunk 4]
```diff
@@ -9586,6 +9586,28 @@ const devices = [
...
+        configure: async (device, coordinatorEndpoint) => {
+            const endpoint = device.getEndpoint(1);
+            await bind(endpoint, coordinatorEndpoint, [
+                'genBasic', 'genPowerCfg', 'genIdentify', 'genTime', 'genPollCtrl', 'hvacThermostat',
+                'hvacUserInterfaceCfg',
+            ]);
+            await configureReporting.thermostatTemperature(endpoint);
+            await configureReporting.thermostatOccupiedHeatingSetpoint(endpoint);
+            await configureReporting.thermostatPIHeatingDemand(endpoint);
+           },
```
Comment (sunlix) [inline devices.js:9608]: remove 3 whitespaces before closing curly brace.

**Comment by Koenkk** [comment]
Thanks and congratz on your first PR.

**Pull request merged**